trio_websocket/_impl.py

```diff
@@ -615,7 +644,8 @@ def __init__(self, stream, wsproto, *, host=None, path=None,
         self._reader_running = True
         if wsproto.client:
             self._initial_request = Request(host=host, target=path,
-                subprotocols=subprotocols)
+                subprotocols=subprotocols or [],
```

Comment:
Why "or []" if these accept None?

Reply:
Wsproto 0.14.0 has a funky way of handling defaults, where the absence of a keyword argument is handled differently than setting that argument to `None`. It turns out that I can leave off `or []` for `subprotocols` but not for `extra_headers`.

(This default values bug is resolved on wsproto master by converting events to dataclasses, but consuming that change here will force us to drop support for Python 3.5.)